src/new-components/button-link/button-link.tsx

```diff
@@ -1,41 +1,44 @@
 import * as React from 'react'
-import { Box } from '../box'
 import { polymorphicComponent } from '../../utils/polymorphism'
+import { BaseButton } from '../base-button'
```

Comment:
A few questions:

- What do you do in that case when wanting to render a react router link, which does not use `href` but uses the prop `to` instead?
- How to you approach type safety on the props?
    - For instance, `button` has a `type` attribute which has to be one of `"button"`, `"submit"` or `"reset"`. But `a` tags have a `type` attribute with an entirely different meaning and possible values.
    - Anchor tags may have `rel` and `target` attributes. And `name`, and `download`, and more. This is just the tip of the iceberg.

I think that approach may work well in a pure js environment, but with types involved it may get messy. Even the ref element type could get messy.

Anyway, I'll explore it early next week. Maybe it's not a big deal, even with these concerns in mind.

Reply:
> What do you do in that case when wanting to render a react router link, which does not use href but uses the prop to instead?

Looking at Ionic's implementation, they use the `href` prop for a route and have a `routerDirection` prop to handle the behaviour. Could we do something similar?

* https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/button/button.tsx#L159
* https://github.com/ionic-team/ionic-framework/tree/main/core/src/components/button#properties

> How to you approach type safety on the props?

We might be able to borrow inspiration from the Polaris implementation [here](https://github.com/Shopify/polaris-react/blob/main/src/components/Button/Button.tsx#L74).

Reply:
A quick check on the code of a couple of these libraries sheds some light on issues that they can get away with that I cannot.

For instance,

- Polaris ignores forwarding refs altogether. None of the involved components use `React.forwardRef` ([example](https://github.com/Shopify/polaris-react/blob/1602da1611828b9fabe6a4c3e17a7eb32c628f72/src/components/UnstyledButton/UnstyledButton.tsx#L15)). This makes it so that they do not have to worry about properly typing the `ref` dynamically based on the presence of `href` (`url` in their case).
- material-ui writes all their code in pure js, and then provide type definitions. I suspect this also simplifies the code part of things.
- ionic's code is even more inscrutable, using decorators and react classes.

I'm afraid I'm not going to get through with it. I'm sure there are ways in which to achieve it. We can always deprecate the `ButtonLink` in the future if we figure it out while keeping all the bells and whistles. I followed here Braid's approach of having two separate components for this.

Reply:
That's fair @gnapse, I appreciate you looking into the feasibility here 👏 

Once the height/padding comment is addressed this PR is good to go as far as I'm concerned 👍